src/test/compile-middleware_test.ts

```diff
@@ -109,4 +109,32 @@ suite('compile-middleware', () => {
       assert.notInclude(response.text, `<script>\nclass A {}\n</script>`);
     });
   });
+
+  test('needCompilation', () => {
+    const cases: {userAgent: string, expected: boolean}[] = [
+      {
+        userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/14.99999',
+        expected: true,
+      },
+      {
+        userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/15.14986',
+        expected: true,
+      },
+      {
+        userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/15.15063',
+        expected: false,
+      },
+      {
+        userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/16.00000',
+        expected: false,
+      },
+    ];
+    for (const c of cases) {
+      assert.equal(needCompilation(c.userAgent), c.expected, c.userAgent);
```

Comment:
I find one downside of assertions/tests in plain loops to be that the line the error occurred on doesn't clearly indicate the failure. In this case the UA string is the only state needed, so it'd be easy to look up and find the test data, but also in this case I'm not sure the object representation above is more compact than an inline assertion.

No need to change though.

Also, just to be self-contradictory, another alternative I like is custom assertions (though I generally avoid Chai's recommended global modifying approach of `Assertion.addMethod()`):

```typescript
function assertNeedsCompilation(ua: string, expectation: boolean) {
  assert.equal(needCompilation(ua), expectation, ua);
}
assertNeedsCompilation('Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/14.99999', true);
```

Some of the same downsides as a loop, unless there's a way to black-box the assertion function I don't know about.